### PR TITLE
Feature/argocd to spokes

### PR DESCRIPTION
**Pull request by markoskandylis**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: setup-git.sh [hunk 1]
```diff
@@ -15,26 +15,33 @@ mkdir -p ${GITOPS_DIR}
...
 
 git clone ${gitops_workload_url} ${GITOPS_DIR}/apps
 mkdir ${GITOPS_DIR}/apps/backend
 touch ${GITOPS_DIR}/apps/backend/.keep
 mkdir ${GITOPS_DIR}/apps/frontend
 touch ${GITOPS_DIR}/apps/frontend/.keep
 git -C ${GITOPS_DIR}/apps add . || true
-git -C ${GITOPS_DIR}/apps commit -m "initial commit" || true
-git -C ${GITOPS_DIR}/apps push  || true
+git -C ${GITOPS_DIR}/apps commit -m "initial commit" --no-verify || true
```
Comment (allamand) [inline setup-git.sh:26]: Why add —no-verify ?

File: gitops/fleet/charts/fleet-argo/values-test.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+name: myteams
```
Comment (allamand) [inline gitops/fleet/charts/fleet-argo/values-test.yaml:1]: What are those teams ? We deploy application for the web store website ?

File: gitops/fleet/charts/fleet-argo/values-test.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+annotations:
+  toplevel: annotations
+labels:
+  toplevel: labels
+
+namespaces:
+  namespace1:
```
Comment (allamand) [inline gitops/fleet/charts/fleet-argo/values-test.yaml:9]: What is namespace1

File: gitops/fleet/charts/fleet-argo/values.yaml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  ingress:
+    deny:
+      all:
+        # -- create a NetworkPolicy that denies all incoming traffic
+        enabled: true
+        # -- define a PodSelector that should be used. Uses all Pods by default
+        podSelector: {}
+
+
+# namespaces:
```
Comment (allamand) [inline gitops/fleet/charts/fleet-argo/values.yaml:35]: What is all that comments section ?

File: gitops/fleet/fleet/argoproject-values.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# using upstream argo chart https://github.com/argoproj/argo-helm/tree/main/charts/argocd-apps
+projects:
+- name: argo-admin
+  sourceRepos:
+  - 'tobereplace'
```
Comment (allamand) [inline gitops/fleet/fleet/argoproject-values.yaml:5]: Is it place holder ?

File: gitops/fleet/fleet/argoproject-values.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  namespace: argocd
+  additionalLabels: {}
+  additionalAnnotations: {}
+  description: Argo Admin
```
Comment (allamand) [inline gitops/fleet/fleet/argoproject-values.yaml:9]: Should this be dynamic depending on the fleet cluster name itself ?

File: setup-git.sh [hunk 7]
```diff
@@ -15,26 +15,33 @@ mkdir -p ${GITOPS_DIR}
 gitops_workload_url="$(terraform -chdir=${ROOTDIR}/terraform/codecommit output -raw gitops_workload_url)"
 gitops_platform_url="$(terraform -chdir=${ROOTDIR}/terraform/codecommit output -raw gitops_platform_url)"
 gitops_addons_url="$(terraform -chdir=${ROOTDIR}/terraform/codecommit output -raw gitops_addons_url)"
+gitops_fleet_url="$(terraform -chdir=${ROOTDIR}/terraform/codecommit output -raw gitops_fleet_url)"
```
Comment (allamand) [inline setup-git.sh:18]: You remove workloads elsewhere but not here. Not sur how fleet replace workloads ?

File: terraform/hub/main.tf [hunk 8]
```diff
@@ -48,15 +48,10 @@ locals {
   gitops_addons_path     = data.terraform_remote_state.git.outputs.gitops_addons_path
   gitops_addons_revision = data.terraform_remote_state.git.outputs.gitops_addons_revision
 
-  gitops_platform_url      = data.terraform_remote_state.git.outputs.gitops_platform_url
-  gitops_platform_basepath = data.terraform_remote_state.git.outputs.gitops_platform_basepath
-  gitops_platform_path     = data.terraform_remote_state.git.outputs.gitops_platform_path
-  gitops_platform_revision = data.terraform_remote_state.git.outputs.gitops_platform_revision
-
-  gitops_workload_url      = data.terraform_remote_state.git.outputs.gitops_workload_url
```
Comment (allamand) [inline terraform/hub/main.tf:56]: We keep workload or not seems inconsistent in the PR

File: terraform/hub/main.tf [hunk 9]
```diff
@@ -129,22 +126,16 @@ locals {
       addons_repo_revision = local.gitops_addons_revision
     },
     {
-      platform_repo_url      = local.gitops_platform_url
```
Comment (allamand) [inline terraform/hub/main.tf:132]: We don’t need platform in hub ?

**Review by allamand — CHANGES_REQUESTED** [review]
Nice work, few questions

File: setup-git.sh (hunk 1 above)
Comment (markoskandylis) [reply setup-git.sh:26]: Yes i need to remove that good catch

File: terraform/hub/main.tf [hunk 10]
```diff
@@ -48,15 +48,10 @@ locals {
   gitops_addons_path     = data.terraform_remote_state.git.outputs.gitops_addons_path
   gitops_addons_revision = data.terraform_remote_state.git.outputs.gitops_addons_revision
 
-  gitops_platform_url      = data.terraform_remote_state.git.outputs.gitops_platform_url
```
[the remark this answers is not in the document]
Comment (markoskandylis) [reply terraform/hub/main.tf:51]: Hub will only push fleet helm charts to spokes and we keep addons for the hub deployments. But this is not the final version the gitops-bridge part is still worked on let catch up next week to share the idea behind it :)

File: gitops/fleet/charts/fleet-argo/values-test.yaml (hunk 2 above)
Comment (markoskandylis) [reply gitops/fleet/charts/fleet-argo/values-test.yaml:1]: That will be removed as well on the v2 gitops-bridge it was just for testing puerposes

**Pull request merged**